src/core/Global.ttslua

```diff
@@ -303,12 +303,36 @@ function drawChaosToken(params)
 
     -- add the token to the list, compute new position based on list length
     tokenOffset[1] = tokenOffset[1] + (0.17 * #chaosTokens)
-    local token = chaosBag.takeObject({
-      index = 0,
-      position = params.mat.positionToWorld(tokenOffset),
-      rotation = params.mat.getRotation()
-    })
+    local tokenIndex = 0
+    
+    if params.sealedToken then -- resolve a sealed token from a card
+     
+      token = getObjectFromGUID(params.sealedToken)
+      token.setPositionSmooth(params.mat.positionToWorld(tokenOffset))
+      local guid = token.getGUID()
+      local type = token.getName()
```

Comment:
Nitpick: This variable should not be called type (either `name` or `tokenType` would be better imo)